src/server/run_sched.c

```diff
@@ -155,6 +158,190 @@ put_sched_cmd(int sock, int cmd, char *jobid)
 	return -1;
 }
 
+/**
+ * @brief
+ * 		find_assoc_sched - find the corresponding scheduler which is responsible
+ * 		for handling this job.
+ *
+ * @param[in]	jid	- job id
+ * @param[out]target_sched	- pointer to the corresponding scheduler to which the job belongs to
+ *
+ * @retval - 1 if success
+ * 	   - 0 if fail
+ */
+int
+find_assoc_sched_jid(char *jid, pbs_sched **target_sched)
+{
+	job *pj;
+	pbs_queue *pq;
+	attribute *part_attr;
+	int k;
+	char *q_name;
+	pbs_sched *psched;
+	int t;
+
+	*target_sched = NULL;
+
+	t = is_job_array(jid);
+	if ((t == IS_ARRAY_NO) || (t == IS_ARRAY_ArrayJob))
+		pj = find_job(jid);		/* regular or ArrayJob itself */
+	else
+		pj = find_arrayparent(jid); /* subjob(s) */
+
+	if (pj == NULL)
+		return 0;
+
+	q_name = pj->ji_qs.ji_queue;
+	pq = find_queuebyname(q_name);
+	if (pq == NULL)
+		return 0;
+
+	if (pq->qu_attr[QA_ATR_partition].at_flags & ATR_VFLAG_SET) {
+		psched = (pbs_sched*) GET_NEXT(svr_allscheds);
+		while (psched) {
+			part_attr = &(psched->sch_attr[SCHED_ATR_partition]);
+			if (part_attr->at_flags & ATR_VFLAG_SET) {
+				for (k = 0; k < part_attr->at_val.at_arst->as_usedptr; k++) {
+					if ((part_attr->at_val.at_arst->as_string[k] != NULL)
+							&& (!strcmp(part_attr->at_val.at_arst->as_string[k],
+									pq->qu_attr[QA_ATR_partition].at_val.at_str))) {
+						*target_sched = psched;
+						return 1;
+					}
+				}
+			}
+			psched = (pbs_sched*) GET_NEXT(psched->sc_link);
+		}
+	} else {
+		*target_sched = dflt_scheduler;
+		return 1;
+	}
+	return 0;
+}
+
+/**
+ * @brief
+ * 		find_assoc_sched - find the corresponding scheduler which is responsible
+ * 		for handling this job.
+ *
+ * @param[in]	jid	- job id
+ * @param[out]target_sched	- pointer to the corresponding scheduler to which the job belongs to
+ *
+ * @retval - 1 if success
+ * 	   - 0 if fail
+ */
```

Comment:
I think you don't need this function. You can just use find_assoc_sched_partition() and pass pj->ji_qs.ji_queue to it

Reply:
Implemented.